.github/workflows/main.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  pull_request:
+    branches: 
+      - master
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
```

Comment:
could we do this on RHEL8 instead?

Reply:
It's a GitHub thing, your options are mac, windows, or ubuntu :slightly_frowning_face: 

I could look into updating it to run in a rhel docker container, but this is my first go at configuring GitHub actions.

Reply:
+1 for this PR, +2 if you want to do a followup.